src/registrar/models/utility/domain_helper.py

```diff
@@ -29,19 +29,19 @@ def validate(cls, domain: str | None, blank_ok=False) -> str:
         if not isinstance(domain, str):
             raise ValueError("Domain name must be a string")
         domain = domain.lower().strip()
-        if domain == "":
-            if blank_ok:
-                return domain
-            else:
-                raise errors.BlankValueError()
+        if domain == "" and not blank_ok:
+            raise errors.BlankValueError()
         if domain.endswith(".gov"):
             domain = domain[:-4]
         if "." in domain:
             raise errors.ExtraDotsError()
         if not DomainHelper.string_could_be_domain(domain + ".gov"):
             raise ValueError()
-        if not check_domain_available(domain):
-            raise errors.DomainUnavailableError()
+        try:
+            if not check_domain_available(domain):
+                raise errors.DomainUnavailableError()
+        except Exception:
```

Comment:
**non-blocking Suggestion:** always tried to avoid using a generic Exception. This is because you (and others maintaining the code) should know what type of errors can result from a function. Here it should be a RegistryError correct? In which case, I would catch a RegistryError instead. Double check that is the correct error

Reply:
Ah got it - I ended up catching RegistryError and also saving using as/from err - thank you both for the explanations and suggestions!